bin/import_xml2obd.py

```diff
@@ -0,0 +1,179 @@
+
+
+def title(k,v):
+    if 'English' in k:
+        sub_key='en'
+    elif 'French' in k:
+        sub_key='fr'
+    else:
+        raise Exception('no such desc')
+    return 'title_translated', {sub_key:v}
+
+organizations = {}
+
+
+def read_organizations_from_json(filename):
+    with open(filename) as f:
+        for line in f:
+            rec = json.loads(line)
+            title = rec['title'].split('|')[0].strip()
+            organizations[title] = rec['id']
+    assert(len(organizations)>100)
+
+
+def owner_org(k,v):
+    v = v.split('|')[0].strip()
+    id = organizations.get(v, None)
+    if not id:
+        raise Exception('No such organiztion ' + v)
+    return 'owner_org', id
+
+
+def _get_choices_value(preset, val):
+    name = preset['field_name']
+    res = []
+    for item in preset['choices']:
+        for label in item['label'].values():
+            if label in val:
+                res.append(item['value'])
+                break
+    return name, res
+
+xml2obd=OrderedDict([
+    ('Description English', desc),
+    ('Description French', desc),
+    ('Title English', title),
+    ('Title French', title),
+    ('Audience', lambda k,v: _get_choices_value(audience, v),),
+    ('Subject', lambda k,v: _get_choices_value(canada_subject, v),),
+    ('Publisher Organization', owner_org),
+
+    ('Date Created', lambda k, v: ('metadata_created',v)),
+    ('Date Modified', lambda k, v: ('metadata_modified',v)),
+    ('Classification Code', lambda k, v: ('doc_classification_code',v)),
+    ('Creator', lambda k, v: ('author',v)),
+    ('retentionTrigger', None),
+    ('retentionPeriod', None),
+    ('Status Date', None),
+])
+
+xml2resource={
+    'Resource Type': lambda k,v: _get_choices_value(canada_resource_type, v),
+    'Record Date': lambda k,v: ('created', v),
+    'Unique ID': lambda k,v: ('unique_identifier', v),
+    'Language': lambda k,v: _get_choices_value(canada_resource_language, v),
+}
+
+
+def xml_obd_mapping(dict_data, map_dict):
+    res = {}
+    for k,f in map_dict.iteritems():
+        if k in dict_data:
+            v = dict_data[k]
+            dict_data.pop(k)
+            if not f:
+                continue
+            new_key, new_val = f(k,v)
+            val = res.get(new_key, None)
+            if isinstance(val,dict):
+                val.update(new_val)
+            elif isinstance(val, list):
+                for _val in new_val:
+                    val.append(_val)
+            elif not new_val:
+                pass
+            else:
+                res[new_key] = new_val
+    return res
+
+
+def get_preset(name):
+    for it in presets['presets']:
+        if it['preset_name']==name:
+            return it['values']
+    return None
+
+
+def main():
+    global presets, audience, canada_resource_type,canada_subject
+    global canada_resource_language
+    refs, extras = read_xml(sys.argv[1])
+    read_organizations_from_json(sys.argv[2])
+    presets = read_presets(sys.argv[3])
```

Comment:
slightly better than this would be calling the API for `scheming_dataset_schema_show(type='doc')` because some of our fields might get overridden. Not worth the change for now, just wanted to make a note.